### DISCUSSION TITLE
platform.io integration?

### DISCUSSION THREAD

**Discussion by robmaxtech** [SpenceKonde/DxCore#110] [Q&A]
Hello all! I just did the jump over to Visual Studio Code, together with platform.io. Before, I was working with Visual Studio and the VisualMicro extension for AVR uC projects, but VS Code is so much more lightweight, opensource, incredibly extensible, and with platform.io you get a unified platform for tons of microcontrollers. All with perfect code completion/Intellisense and so on. I learnt about it by MCUdude who wrote the MEgaCoreX https://github.com/MCUdude/MegaCoreX for all the new ATMEGAs, and his core integrates perfectly into this environment. **BTW: You, Spence, and MCUdude are my absolute heroes!** 
Now I wonder if the DxCore could be integrated also into VScode/platform.io; I think yes, but I'm not good figuring out how this stuff works, and unfortunately my time is quite limited for this too. So maybe there is somebody giving some pointers how to do this? Thanks a lot, guys!

**Comment by SpenceKonde** [comment]
There is an open issue for this over there/. Arduino hasn't put out a toolchain package to support recent parts. They seem to have forgotten how or something. I've been hacking them together to support my core (I do a build with new atpacks for one platform, but use the old executable binaries and repack it). I've told them to go ahead and use mine, just host it themselves so I don't get socked with bandwidth bills) and I thought they were going to do it but nothing ever comes of it; It needs to be done by someone on their team though not by me. In any event I have my hands way too full I cannot take on anything else including nagging the platform.io people, I'm 4 days behind where I thought I would be now, I should have had 1.3.6 of DxCore out at the start of the weekend, but instead have been working on this solid since then.

**Reply by robmaxtech** [reply]
Ok, so the platform.io guys have to do it... So I will nag them too! Thanks for the speedy reply - keep up your awesome work!

**Comment by robmaxtech** [comment]
News about this topic: Accidentally, I found this: https://github.com/brunob45/platform-atmelavrdx
It's a ready-to-use platform for platform.io that integrates the DxCore - big thanks to Bruno B. who did the job.
Installation: $ platformio platform install https://github.com/brunob45/platform-atmelavrdx, or via GUI in VC Code:
PIO Home - Platforms - Advanced Installation - enter the URL and "Install".
